### PR TITLE
Update the SearchBar to make use of the useApp hook to retrieve the search icon specified in the createApp call

### PR DESCRIPTION
**Pull request by RikClaessens**
## Hey, I just made a Pull Request!

The goal of this PR is to make use of the useApp hook in the SearchBar to retrieve the specified custom search icon provided to the Backstage `createApp` function. While implementing this change I noticed that all test cases of SearchBar as well as SearchAutocomplete were failing. This was due the test cases being wrapped by renderWithEffects instead of renderInTestApp. Finally, in order to test the implementation of the new search icon, I added the `icons` option to the `renderInTestApp` function. With this change I can verify that the custom search icon gets rendered when it is specified from `createApp`.

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Added or updated documentation
- [X] Tests for new functionality and regression tests for bug fixes
- [X] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

image
In the screenshot you're able to notice the Backstage icon now appearing in the search bar instead of the magnifying glass icon.

### CODE REVIEW

**Comment by RikClaessens** [comment]
Awaiting review, so I wouldn't consider the PR to be stale. Please let me know if the PR is OK or if any changes are required. Thanks!

File: .changeset/sixty-rabbits-cheat.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-search-react': minor
+---
+
+Make use of the useApp hook to retrieve the specified search icon in the SearchBar
```
Comment (awanlin) [inline .changeset/sixty-rabbits-cheat.md:5]:
```suggestion
Make use of the `useApp` hook to retrieve the specified search icon in the SearchBar
```

Same here, that should make Vale happy

**Review by awanlin — CHANGES_REQUESTED** [review]
Thanks for this @RikClaessens, just a pair of comments that should help resolve the Vale issues.

**Comment by RikClaessens** [comment]
Thanks @awanlin for your comments, I've updated the PR with your suggestions.

**Review by awanlin — APPROVED** [review]
Thanks for the contribution @RikClaessens 🚀 

Just needs a review form the Maintainers now 👍

**Review by freben — APPROVED** [review]

**Pull request merged**